src/components/core/NumberField.js

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { InputField } from '@dhis2/ui';
+import styles from './styles/InputField.module.css';
+
+// Adds support for min/max values for @dhis2/ui InputField
+const NumberField = ({ label, value, min, max, onChange, style }) => {
+    const onNumberChange = useCallback(
+        ({ value }) => {
+            if (
+                (min === undefined || value >= min) &&
+                (max === undefined || value <= max)
+            ) {
+                onChange(value);
+            }
+        },
+        [min, max]
+    );
+
+    return (
+        <div className={styles.inputField} style={style}>
+            <InputField
+                type="number"
+                label={label}
+                value={String(value)}
+                onChange={onNumberChange}
+            />
+        </div>
+    );
+};
+
+NumberField.propTypes = {
```

Comment:
The proptypes are not in order. Not too concerned about that actually but curious why eslint hasn't picked up on it? It usually does in DV and Analytics at least.

Reply:
Not sure either, but the proptypes should now align with the input props.